### PR TITLE
Customer service branch

### PR DESCRIPTION
**Pull request by KajalMathur**
Kindly review my code.

### CODE REVIEW

File: src/main/java/com/customer/address/Address.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@NoArgsConstructor
+@AllArgsConstructor
+@Getter
+@Setter
+@Builder
+@Entity
+@Table(name = "Address")
```
Comment (AakashIsTheLimit) [inline src/main/java/com/customer/address/Address.java]: why do you need this ?

**Review by AakashIsTheLimit — CHANGES_REQUESTED** [review]

File: src/main/java/com/customer/address/Address.java (hunk 1 above)
Comment (KajalMathur) [reply src/main/java/com/customer/address/Address.java]: Removed

File: src/main/java/com/customer/CustomerResponse/CustomerResponse.java [hunk 2]
```diff
@@ -3,6 +3,8 @@
 import java.util.Calendar;
 import java.util.Date;
 
+import com.customer.address.Address;
+
```
Comment (AakashIsTheLimit) [inline src/main/java/com/customer/CustomerResponse/CustomerResponse.java:7]: Use @data here instead

File: src/main/java/com/customer/CustomerResponse/CustomerResponse.java [hunk 3]
```diff
@@ -15,12 +17,17 @@
...
 	private String lastName;
-	private String address;
+	private Address address;
 	private Date joiningDate;
 	private Date expiryDate;
-	private String status;
+	Status status;
+
+	private enum Status {
+		New, Experienced, Expired, Invalid;
```
Comment (AakashIsTheLimit) [inline src/main/java/com/customer/CustomerResponse/CustomerResponse.java:26]: Generally we have it in all CAPS

File: src/main/java/com/customer/controller/CustomerController.java [hunk 4]
```diff
@@ -6,37 +6,50 @@
...
 	public String createCustomer(@Valid @RequestBody Customer customer) {
-
 		customerServiceImpl.createCustomer(customer);
 		return "Customer is created successfully";
 
 	}
 
+	/* get the customer by Id */
+	@GetMapping("/customers/{id}")
+	public @ResponseBody Object findCustomerById(@PathVariable int id) {
```
Comment (AakashIsTheLimit) [inline src/main/java/com/customer/controller/CustomerController.java:50]: why return object ?

File: src/main/java/com/customer/service/CustomerServiceImpl.java [hunk 5]
```diff
@@ -9,69 +10,60 @@
...
+					.firstName(cus.get().getFirstName()).lastName(cus.get().getLastName())
+					.joiningDate(cus.get().getJoiningDate()).expiryDate(cus.get().getExpiryDate())
+					.address(cus.get().getAddress()).build();
+			customerResponse.setStatus(cus.get().getJoiningDate(), cus.get().getExpiryDate());
+			return customerResponse;
+		}
 
-	}
+		else
+			return "No data found for the requested user";
```
Comment (AakashIsTheLimit) [inline src/main/java/com/customer/service/CustomerServiceImpl.java:65]: throw error here instead

File: src/test/java/com/customer/customerService/CustomerServiceApplicationTests.java [hunk 6]
```diff
@@ -1,16 +1,75 @@
...
+	@Mock
+	CustomerRepository customerRepository;
+
+	@InjectMocks
+	CustomerServiceImpl customerServiceImpl;
+
+	@Test
+	public void postRequestTest() {
+		// Given
+		Address address = Address.builder().streetName("Abc").state("Haryana").country("France").build();
```
Comment (AakashIsTheLimit) [inline src/test/java/com/customer/customerService/CustomerServiceApplicationTests.java:37]: can you add couple of more asserts to the unit tests

File: src/main/java/com/customer/service/CustomerServiceImpl.java [hunk 7]
```diff
@@ -9,69 +10,60 @@
...
-	public void updateCustomer(Customer cust) {
-		// TODO Auto-generated method stub
-
+		Collections.sort(custRespList, new SortingComparator());
+		return custRespList;
 	}
 
 	@Override
-	public void deleteAllCustomers() {
+	public Object findCustomerById(int id) {
```
Comment (AakashIsTheLimit) [inline src/main/java/com/customer/service/CustomerServiceImpl.java:52]: Use @ControllerAdvise and learn Exceptional handling in spring boot

**Review by AakashIsTheLimit — CHANGES_REQUESTED** [review]

File: src/main/java/com/customer/CustomerResponse/CustomerResponse.java (hunk 2 above)
Comment (KajalMathur) [reply src/main/java/com/customer/CustomerResponse/CustomerResponse.java:7]: Done

File: src/main/java/com/customer/CustomerResponse/CustomerResponse.java (hunk 3 above)
Comment (KajalMathur) [reply src/main/java/com/customer/CustomerResponse/CustomerResponse.java:26]: Done

File: src/main/java/com/customer/controller/CustomerController.java (hunk 4 above)
Comment (KajalMathur) [reply src/main/java/com/customer/controller/CustomerController.java:50]: Changes has done.

File: src/main/java/com/customer/service/CustomerServiceImpl.java (hunk 5 above)
Comment (KajalMathur) [reply src/main/java/com/customer/service/CustomerServiceImpl.java:65]: Exception has handled by using the @ControllerAdvice and @ExceptionHandler annotations of spring.

File: src/main/java/com/customer/service/CustomerServiceImpl.java (hunk 7 above)
Comment (KajalMathur) [reply src/main/java/com/customer/service/CustomerServiceImpl.java:52]: done

File: src/test/java/com/customer/customerService/CustomerServiceApplicationTests.java (hunk 6 above)
Comment (KajalMathur) [reply src/test/java/com/customer/customerService/CustomerServiceApplicationTests.java:37]: Done , here i am mocking two classes.

Comment (KajalMathur) [reply src/test/java/com/customer/customerService/CustomerServiceApplicationTests.java:37]: Done

**Comment by KajalMathur** [comment]
Changes has done.

**Pull request closed without merging**